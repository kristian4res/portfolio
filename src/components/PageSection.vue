<script setup lang="ts">
import { defineProps } from 'vue'
import { Divider } from 'primevue'

defineProps({
  heading: {
    type: String,
    required: true,
  },
})
</script>

<template>
  <section class="flex-col flex">
    <h1 class="font-bold text-2xl">{{ heading }}</h1>
    <Divider />
    <div class="text-lg">
      <slot></slot>
    </div>
  </section>
</template>
