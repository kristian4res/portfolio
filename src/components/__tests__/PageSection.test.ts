import { mount } from '@vue/test-utils'
import { describe, it, expect } from 'vitest'
import PageSection from '../PageSection.vue'

describe('PageSection.vue', () => {
  it('matches snapshot', () => {
    const wrapper = mount(PageSection, {
      props: {
        heading: 'Hello, world!',
      },
      slots: {
        default: 'This is a test slot content',
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
