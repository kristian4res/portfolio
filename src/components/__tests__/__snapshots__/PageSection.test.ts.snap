// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageSection.vue > matches snapshot 1`] = `
"<section class="flex-col flex flex-wrap">
  <h1 class="font-bold text-2xl mb-4">Hello, world!</h1>
  <div class="text-lg sm:text-wrap">This is a test slot content</div>
  <div class="p-divider p-component p-divider-horizontal p-divider-solid p-divider-left" style="justify-content: center;" role="separator" aria-orientation="horizontal" data-pc-name="divider" pc1="" data-pc-section="root">
    <!---->
  </div>
</section>"
`;
